src/top-bar/top-bar.js

```diff
@@ -8,7 +9,9 @@ const TopBar = () => (
     <>
         <WorkflowSelect />
         <PeriodSelect />
-        <OrgUnitSelect />
+        <ApprovalStatusesProvider>
```

Comment:
This provider could be moved up to the top level, which would allow the status in the bottom bar to use it to fetch and render the approval status of the current selection. Doing so would require changing the shape of the approval statuses map from `(workflow, period) => status` to `(workflow, period) => { status, approvedAt, approvedBy }`.

One benefit of using the provider in this way is that it would make an eventual redux refactor easier.

Reply:
Yep our two options are:
1. Invalidate the cache every time the org unit select is opened
2. Update org unit status in the bottom bar

Reply:
Agreed. However, I feel that if we were to go with option 1 we'd lose most of the benefits this solution brings to the table.